core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -1930,8 +1930,13 @@ public RelBuilder filter(Iterable<CorrelationId> variablesSet,
     if (config.simplify()) {
       conjunctionPredicates = simplifier.simplifyFilterPredicates(predicates);
     } else {
+      List<RexNode> simplified = new ArrayList<>();
+      for (RexNode predicate : predicates) {
+        RexNode simple = RexSimplify.simplifyComparisonWithNull(predicate, getRexBuilder());
```

Comment:
When `config.simplify()` returns false, `simplifyComparisonWithNull` is still called. Does `simplifyComparisonWithNull` count as a simplification process? If it does, the logic seems very odd. If it doesn't, it appears that `simplifyComparisonWithNull` might return `False` under certain conditions. Additionally, a List variable named "simplified" is created here, which doesn't seem easy to understand.